api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -120,10 +135,12 @@ func (instance *OpenStackDataPlaneRole) InitConditions() {
 	}
 
 	haveCephSecret := false
-	for _, extraMount := range instance.Spec.NodeTemplate.ExtraMounts {
-		if extraMount.ExtraVolType == "Ceph" {
-			haveCephSecret = true
-			break
+	for _, node := range instance.Spec.NodeTemplate.Nodes {
+		for _, extraMount := range node.ExtraMounts {
```

Comment:
Should this use the inherited value? I'm not understanding how this works.

Reply:
We used to have `nodeTemplate/extraMounts` for all nodes in the role. With this it looks like we'll have `nodeTemplate/Nodes/extraMounts`. Does `Nodes` refer to the nodeset? If so, then I don't need to then define it per node since it would apply to multiple nodes in the nodeset?

To be sure, the PR could include an update of `config/samples/dataplane_v1beta1_openstackdataplane_ceph.yaml` which uses `extraMounts`.

https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplane_ceph.yaml#L78

You don't actually have to use ceph to test it. You could just stick some arbitrary data in the ceph secret:
```
KEY=foo
CONF=bar
```
and then follow these two steps:

- https://github.com/openstack-k8s-operators/docs/blob/main/ceph.md#create-a-ceph-secret
- https://github.com/openstack-k8s-operators/docs/blob/main/ceph.md#access-the-ceph-secret-via-extramounts

Then if the ansibleEE pod can mount the data in the secret everything should be fine. If the EDPM nodes gets a file called `/etc/ceph.conf` containing only `bar` after ansible runs, then it worked.

I could help you with testing that if you like